src/combinatorics/composition.lean

```diff
@@ -381,31 +378,198 @@ begin
+This makes sense mostly when `n = l.length`, but this is not necessary for the definition. -/
+def split_wrt_composition (l : list α) (c : composition n) : list (list α) :=
+split_wrt_composition_aux l c.blocks
+
+local attribute [simp] split_wrt_composition_aux.equations._eqn_1
+
+local attribute [simp]
+lemma split_wrt_composition_aux_cons (l : list α) (n ns) :
+  l.split_wrt_composition_aux (n :: ns) = take n l :: (drop n l).split_wrt_composition_aux ns :=
+by simp [split_wrt_composition_aux]
+
+lemma length_split_wrt_composition_aux (l : list α) (ns) :
+  length (l.split_wrt_composition_aux ns) = ns.length :=
+by induction ns generalizing l; simp *
+
+/-- When one splits a list along a composition `c`, the number of sublists thus created is
+`c.length`. -/
+@[simp] lemma length_split_wrt_composition (l : list α) (c : composition n) :
+  length (l.split_wrt_composition c) = c.length :=
+length_split_wrt_composition_aux _ _
+
+lemma map_length_split_wrt_composition_aux {ns : list ℕ} :
+  ∀ {l : list α}, ns.sum ≤ l.length → map length (l.split_wrt_composition_aux ns) = ns :=
+begin
+  induction ns with n ns IH; intros l h; simp at h ⊢,
+  have := le_trans (nat.le_add_right _ _) h,
+  rw IH, {simp [this]},
+  rwa [length_drop, nat.le_sub_left_iff_add_le this]
+end
+
+/-- When one splits a list along a composition `c`, the lengths of the sublists thus created are
+given by the block sizes in `c`. -/
+lemma map_length_split_wrt_composition (l : list α) (c : composition l.length) :
+  map length (l.split_wrt_composition c) = c.blocks :=
+map_length_split_wrt_composition_aux (le_of_eq c.blocks_sum)
+
+lemma length_pos_of_mem_split_wrt_composition {l l' : list α} {c : composition l.length}
+  (h : l' ∈ l.split_wrt_composition c) : 0 < length l' :=
+begin
+  have : l'.length ∈ (l.split_wrt_composition c).map list.length :=
+    list.mem_map_of_mem list.length h,
+  rw map_length_split_wrt_composition at this,
+  exact c.blocks_pos this
+end
+
+lemma sum_take_map_length_split_wrt_composition
+  (l : list α) (c : composition l.length) (i : ℕ) :
+  (((l.split_wrt_composition c).map length).take i).sum = c.size_up_to i :=
+by { congr, exact map_length_split_wrt_composition l c }
+
+theorem join_split_wrt_composition_aux {ns : list ℕ} :
+  ∀ {l : list α}, ns.sum = l.length → (l.split_wrt_composition_aux ns).join = l :=
+begin
+  induction ns with n ns IH; intros l h; simp at h ⊢,
+  { exact (length_eq_zero.1 h.symm).symm },
+  rw IH, {simp},
+  rwa [length_drop, ← h, nat.add_sub_cancel_left]
+end
+
+/-- If one splits a list along a composition, and then joins the sublists, one gets back the
+original list. -/
+theorem join_split_wrt_composition (l : list α) (c : composition l.length) :
+  (l.split_wrt_composition c).join = l :=
+join_split_wrt_composition_aux c.blocks_sum
+
+/-- If one joins a list of lists, and then splits the join along the right composition, one get
+backs the original list of lists. -/
```

Comment:
```suggestion
/-- If one joins a list of lists and then splits the join along the right composition, one gets
back the original list of lists. -/
```

Reply:
Don't you know the verb "getback", that conjugates as "one getbacks"? :-) Thanks!